Krishvsoni patch 2

krishvsoni opened the pull request:
<!-- Thanks for contributing to Wagtail! 🎉  Please add a description below, explaining the purpose of this pull request - including the issue number of the issue you're fixing (if applicable). -->

Fixes #...

In the README file before installation and setup of the project, there is a line that says to run the module in the virtual enviorment, so in that virtual enviorment i had given official documentation of python virtual enviorment.





_Please check the following:_

-   [ ] Do the tests still pass?[^1]
-   [ ] Does the code comply with the style guide?
    -   [ ] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [✔️] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

<img width="785" alt="image" src="https://github.com/wagtail/wagtail/assets/67964054/3ed8875c-26c7-4fd7-89a3-1f55121c5329">

lb- commented on README.md:
```diff
@@ -48,11 +48,12 @@ Find out more at [wagtail.org](https://wagtail.org/).
 Wagtail works with [Python 3](https://www.python.org/downloads/), on any platform.
 
-To get started with using Wagtail, run the following in a virtual environment:
+To get started with using Wagtail, run the following in a [virtual environment](https://docs.python.org/3/tutorial/venv.html):
```
This may not be exactly right, there are a few different ways to set up a virtual environment.

We may cause more confusion by being specific here. 

I'll let other core team chime in though. We do intentionally keep this brief as there main docs are the place to get full instructions. https://docs.wagtail.org/en/stable/getting_started/tutorial.html#create-and-activate-a-virtual-environment

Maybe we can write

```suggestion
To get started with using Wagtail, run the following in a virtual environment such as [venv](https://docs.python.org/3/tutorial/venv.html):
```

gasman replied:
I prefer @krishvsoni's original version here, and I'd point to the Diataxis framework's advice on tutorials to back this up - [Ignore options and alternatives](https://diataxis.fr/tutorials/#ignore-options-and-alternatives).

Anyone who's knowledgeable / opinionated enough to have a preferred approach to creating virtual environments will happily disregard the link and do their own thing; and for everyone else, knowing that alternatives to venv exist (but not anything about them) doesn't add any value. And, as you say, people who are not familiar with virtual environments at all are better served by the full tutorial.

lb- replied:
Good call, thank you.

@krishvsoni  if you are OK to clean up the other parts of the PR (just revert accidental/unrelated changes), we should be good to go on this one.

lb- commented on README.md:
````diff
@@ -48,11 +48,12 @@ Find out more at [wagtail.org](https://wagtail.org/).
 ![Installing Wagtail](.github/install-animation.gif)
 
 ```sh
+
````
Please remove this accidental whitespace change.

lb- commented on wagtail/api/v2/pagination.py:
```diff
@@ -1,12 +1,9 @@
 from collections import OrderedDict
-
```
Please remove the Changes to this file, I don't think this relates to this PR.

krishvsoni replied:
I made this changes in another branch so here in this branch it's about venv only there is no other change but still I verify and change it.

lb- reviewed (changes requested):
Thanks @krishvsoni - a few bits of feedback and suggestions for changes. 

I'm not 100% sure we want this as we do try to keep the Readme brief and generic. However, maybe a slight rewording could help with the link. 

I'll let another core team also review to confirm we are happy with this. 

Thanks for the PR though.

lb- commented:
Please also update the title to be clearer about what you're doing and also remove the 'fixes...' text from the description. 

We have a section about creating a PR in our contribution guide. https://docs.wagtail.org/en/stable/contributing/first_contribution_guide.html#submitting-a-pull-request

lb- reviewed (approved):
Thanks @krishvsoni  I have updated your branch with the changes, I know it's a bit tricky getting used to git branches but I could see you had created this patch-2 branch from your other patch-1 branch.

All good though, fixed up and I will merge this in.

Thanks for helping us improve Wagtail, I have added you to the contributors list as 'Krish Soni' but let me know if you want a different name listed.

krishvsoni commented:
> Thanks @krishvsoni I have updated your branch with the changes, I know it's a bit tricky getting used to git branches but I could see you had created this patch-2 branch from your other patch-1 branch.
> 
> All good though, fixed up and I will merge this in.
> 
> Thanks for helping us improve Wagtail, I have added you to the contributors list as 'Krish Soni' but let me know if you want a different name listed.

Happy to contribute in the Wagtail, I am so grateful to you guys @lb-  and @gasman  for helping me 😊
